### ISSUE TITLE
Crash on startup: "Provided dc_accounts is null, did you unref the accounts object?"

### ISSUE THREAD

**Issue by Hocuri** [deltachat/deltachat-desktop#3076]
- Operating System (Linux/Mac/Windows/iOS/Android):
- Delta Chat Version: 1.43.1
- Steps to reproduce the problem:

I updated DC, installed via flatpak, to the newest available version.

- Screenshots:

![Screenshot_2023-01-08_22-30-35](https://user-images.githubusercontent.com/18012815/211220161-deb13c86-0e4c-442b-aa20-247cc7d480ef.jpg)

- Logs:

```
2023-01-08T21:29:54.577Z	main/index            	INFO	""	"Deltachat Version 1.34.1 flathub 1671910698242"
2023-01-08T21:29:54.937Z	main/index            	INFO	""	"cwd /home/jonathan/.var/app/chat.delta.desktop/config/DeltaChat/accounts"
2023-01-08T21:29:55.048Z	main/index            	CRITICAL	"\n/app/delta/resources/app/src/main/index.ts:108:9"	"Fatal Error during init"	{"stack":"TypeError: Provided dc_accounts is null, did you unref the accounts object?\n    at AccountManager.startJsonRpcHandler (/app/delta/resources/app/node_modules/deltachat-node/node/lib/deltachat.ts:129:13)\n    at DeltaChatController.<anonymous> (/app/delta/resources/app/src/main/deltachat/controller.ts:56:26)\n    at Generator.next (<anonymous>)\n    at fulfilled (/app/delta/resources/app/tsc-dist/main/deltachat/controller.js:28:58)","message":"Provided dc_accounts is null, did you unref the accounts object?"}
```

```
LaunchProcess: failed to execvp:
xdg-settings
LaunchProcess: failed to execvp:
xdg-settings
LaunchProcess: failed to execvp:
xdg-settings
LaunchProcess: failed to execvp:
xdg-settings
LaunchProcess: failed to execvp:
xdg-settings
LaunchProcess: failed to execvp:
xdg-settings
Gtk-Message: 22:39:04.685: Failed to load module "xapp-gtk3-module"
Gtk-Message: 22:39:04.686: Failed to load module "xapp-gtk3-module"
Gtk-Message: 22:39:04.686: Failed to load module "canberra-gtk-module"
Gtk-Message: 22:39:04.686: Failed to load module "canberra-gtk-module"
[2:0108/223904.854218:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0108/223904.854320:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
failed to create accounts: failed to load accounts: failed to create context from file "/home/jonathan/.var/app/chat.delta.desktop/config/DeltaChat/accounts/f58a6c47-c448-417b-bd81-fe87d9d0d9b9/dc.db": failed to run migrations: execute_migration failed for version 93: table sending_domains already exists: Error code 1: SQL error or missing database
ignoring careless call to dc_jsonrpc_init()
ignoring careless call to dc_jsonrpc_next_response()
[57:0108/223905.164420:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
```

- Remarks:
  Not sure if it's a Core or Desktop problem. Esp, is it normal that loading these modules and failing to connect to the bus fails?

**Comment by Simon-Laux** [comment]
> failed to create accounts: failed to load accounts: failed to create context from file "/home/jonathan/.var/app/chat.delta.desktop/config/DeltaChat/accounts/f58a6c47-c448-417b-bd81-fe87d9d0d9b9/dc.db": failed to run migrations: execute_migration failed for version 93: table sending_domains already exists: Error code 1: SQL error or missing database

sounds like a core error

**Comment by Hocuri** [comment]
Is it normal that loading these modules and failing to connect to the bus fails? (`failed to execvp`, `Failed to load module`, `Failed to connect to the bus`)

**Comment by Hocuri** [comment]
I digged into it, yes, it's a core issue. For some reason, one of my accounts actually had the table `sending_domains` existing, but `dbversion` 92. It's an account used purely for testing, not properly configured, with `alice@example.com` as the email address. So, probably most users won't notice this problem.

**Comment by Hocuri** [comment]
I did investigate a bit further since @Simon-Laux said that he had the same bug:

The concerning account had the `sending_domains` table since the beginning of November. The preceeding backup from October doesn't have this account at all, so, seems like I created this account around this time.

The addition of the `sending_domains` table seems to be the only migration that's "already" done, migrations 94-96 are not done yet (e.g. `acpeerstates` doesn't have a `verifier` column).

The `sending_domains` migration was added by me in b1c6c40fa7bd71b1fc1e437c92c2d2f9fed51e30 end October, around the time when I created the account. This could be a sign that I created the account while testing b1c6c40fa7bd71b1fc1e437c92c2d2f9fed51e30 and it already contained this table, and there might be no bug, just a very weird account I created.

Soo, I don't have any idea for this except waiting for @Simon-Laux to find out something about the (maybe even unrelated?) problem on his computer.
